ibm_mq/assets/configuration/spec.yaml

```diff
@@ -105,6 +105,12 @@ files:
           type: array
           items:
             type: string
+      - name: auto_discover_channels
```

Comment:
To reviewer: I used a similar naming scheme as [`auto_discover_queues`](https://github.com/DataDog/integrations-core/blob/ibm_mq-3.13.1/ibm_mq/assets/configuration/spec.yaml#L108), although if these are different scopes then I can change the name

Reply:
I think the naming is fine

Reply:
Maybe mention that these are in addition to the ones provided in `channels` and that it will only report the status but not property metrics (I want to avoid people deciding to just rely on this and not provide anything in `channels` thinking that it is the same)